controllers/storageautoscaler/storageautoscaler_controller.go

```diff
@@ -357,6 +301,15 @@ func (r *StorageAutoscalerReconciler) detectDuplicateStorageAutoScaler(ctx conte
 		if storageAutoScalerItem.Name != storageAutoScaler.Name && storageAutoScalerItem.Spec.DeviceClass == storageAutoScaler.Spec.DeviceClass && storageAutoScalerItem.Spec.StorageCluster.Name == storageAutoScaler.Spec.StorageCluster.Name {
 			err = fmt.Errorf("more than one storage autoscaler present with same device class and same storage cluster name, names are %s and %s", storageAutoScaler.Name, storageAutoScalerItem.Name)
 			r.Log.Error(err, "duplicate cr detected", "device class", storageAutoScaler.Spec.DeviceClass, "storage cluster", storageAutoScaler.Spec.StorageCluster.Name)
+
+			storageAutoScaler.Status.Error = &ocsv1.TimestampedError{
+				Message:   fmt.Sprintf("duplicate cr detected, more than one storage autoscaler present with same device class and same storage cluster name, names are %s and %s delete any one of the autoStorageScaler cr", storageAutoScaler.Name, storageAutoScalerItem.Name),
```

Comment:
nice to include quotes around the names, similar suggestion for other places too in this PR
```suggestion
				Message:   fmt.Sprintf("duplicate cr detected, more than one storage autoscaler present with same device class and same storage cluster name, names are %q and %q delete any one of the autoStorageScaler cr", storageAutoScaler.Name, storageAutoScalerItem.Name),
```